src/draw/drawPolygonOnMask.ts

```diff
@@ -34,31 +36,12 @@ export function drawPolygonOnMask(
     return newMask.drawPolyline([...points, points[0]], otherOptions);
   }
 
-  let polygonMask = Mask.createFrom(newMask);
-
   const filteredPoints = deleteDuplicates(points);
+  const arrayPoints = arrayPointsToObjects(filteredPoints);
 
-  for (let i = 0; i < filteredPoints.length; i++) {
-    const line = lineBetweenTwoPoints(
-      filteredPoints[i],
-      filteredPoints[(i + 1) % filteredPoints.length],
-    );
-    for (let row = 0; row < newMask.height; row++) {
-      for (let column = 0; column < newMask.width; column++) {
-        if (isAtTheRightOfTheLine({ column, row }, line, newMask.height)) {
-          if (polygonMask.getBit(column, row)) {
-            polygonMask.setBit(column, row, 0);
-          } else {
-            polygonMask.setBit(column, row, 1);
-          }
-        }
-      }
-    }
-  }
-
-  for (let row = 0; row < mask.height; row++) {
-    for (let column = 0; column < mask.width; column++) {
-      if (polygonMask.getBit(column, row)) {
+  for (let row = 0; row < newMask.height; row++) {
+    for (let column = 0; column < newMask.width; column++) {
+      if (robustPointInPolygon(arrayPoints, [column, row]) === -1) {
```

Comment:
Could you store the state of the previous pixel (on or off) and if the next pixel has the same state, robustPointInPolygon will yield the same result. This is a low hanging fruit that should improve the speed quite a lot (likely many order of magnitude) before implementing a better algorithm.

Reply:
Yes, I'll do that

Reply:
Actually I don't think this could work, because I don't have the "next pixel state". I am drawing a polygon onto an existing mask and the only data I have about the polygon are it's vertices.

Did I misunderstand what you meant?

Reply:
Ok I see you don't draw first the polygon before filling it. So we will improve this later.

Reply:
Ok I understand what you meant now, I can try to draw the polyline first and then check if the value stays the same.

Reply:
Yes, I'd rather already merge this PR, which is already really big and do that afterwards.

Reply:
I made an issue so that we don't forget to do this: #119